plutus-core/plutus-ir/src/PlutusIR/Transform/LetFloat.hs

```diff
@@ -302,19 +302,23 @@ removeLets t =
          TyAbs a n k tBody  -> TyAbs a n k <$> local (+1) (go tBody)
 
          Let a r bs tIn -> do
-          curDepth <- ask
-          bs' <- forM bs $ \b -> do
-            b' <- b & bindingSubterms go
+          bs' <- forM bs $ \b ->
             if mayHaveEffects b
-            then pure $ Just b' -- keep this let
+            then do
+                -- increment depth since it is an anchor
+                b' <- b & bindingSubterms (local (+1) . go)
+                -- keep this let
+                pure $ Just b'
```

Comment:
Suggest pulling the local out, and using `traverseOf` since I think I always find it a bit confusing to use the traversal directly.
```suggestion
             -- increment depth since it is an anchor
            then local (+1) $ do
                b' <- b & traverseOf bindingSubterms go
                -- keep this let
                pure $ Just b'
```